pkg/notifier/default.go

```diff
@@ -26,22 +25,27 @@ func (d *DefaultNotifier) Push(ctx context.Context, notification common.Notifica
 
 	recipientUser, err := d.userStore.Find(ctx, notification.Recipient())
 	if err != nil {
-		slog.Debug("failed to find user", "id", notification.Context().ID, "user", notification.Recipient().String(), "error", err)
-		return fmt.Errorf("failed to find recipient user: %w", err)
+		level := slog.LevelDebug
+		if !errors.Is(err, user.ErrUserNotFound) {
+			level = slog.LevelError
+		}
+		slog.Log(ctx, level, "failed to find user, will attempt to notify on provided identifiers", "id", notification.Context().ID, "user", notification.Recipient().String(), "error", err)
```

Comment:
is this an acceptable pattern? it's a little odd to have the same log message sometimes fire off as a debug and sometimes as an error. (mainly thinking re: alerting & graphing.)

Reply:
This does feel odd.  Maybe split this into two messages?